src/main/java/io/cryostat/net/web/http/api/beta/AbstractJwtConsumingHandler.java

```diff
@@ -0,0 +1,209 @@
+            Throwable cause = e.getCause();
+            if (cause instanceof SecurityException || cause instanceof SaslException) {
+                ctx.response()
+                        .putHeader(
+                                AbstractAuthenticatedRequestHandler.JMX_AUTHENTICATE_HEADER,
+                                "Basic");
+                throw new ApiException(427, "JMX Authentication Failure", e);
+            }
+            Throwable rootCause = ExceptionUtils.getRootCause(e);
+            if (rootCause instanceof ConnectIOException) {
+                throw new ApiException(502, "Target SSL Untrusted", e);
+            }
+            if (rootCause instanceof UnknownHostException) {
+                throw new ApiException(404, "Target Not Found", e);
+            }
+            throw new ApiException(500, e);
+        } catch (Exception e) {
+            if (e instanceof ApiException) {
+                throw (ApiException) e;
+            }
+            throw new ApiException(500, e);
+        }
+    }
+
+    private JWT validateJwt(RoutingContext ctx)
+            throws ParseException, JOSEException, SocketException, UnknownHostException,
+                    URISyntaxException, MalformedURLException {
+        String token = ctx.queryParams().get("token");
+        JWT parsed;
+        try {
+            parsed = jwt.parseAssetDownloadJwt(token);
+        } catch (BadJWTException e) {
+            throw new ApiException(401, e);
+        }
+
+        URL hostUrl = webServer.get().getHostUrl();
+        URI requestUri = new URI(ctx.request().absoluteURI());
+        URI fullRequestUri =
+                new URI(hostUrl.getProtocol(), hostUrl.getAuthority(), null, null, null)
+                        .resolve(requestUri.getRawPath());
+        URI resourceClaim;
+        try {
+            resourceClaim =
+                    new URI(parsed.getJWTClaimsSet().getStringClaim(AssetJwtHelper.RESOURCE_CLAIM));
+        } catch (URISyntaxException use) {
+            throw new ApiException(401, use);
+        }
+        if (!Objects.equals(fullRequestUri, resourceClaim)) {
+            throw new ApiException(401, "Token resource claim does not match requested resource");
+        }
+
+        try {
+            String subject = parsed.getJWTClaimsSet().getSubject();
+            if (!auth.validateHttpHeader(() -> subject, resourceActions()).get()) {
+                throw new ApiException(401, "Token subject has insufficient permissions");
+            }
+        } catch (ExecutionException | InterruptedException e) {
+            throw new ApiException(401, "Token subject permissions could not be determined");
+        }
+
+        return parsed;
+    }
+
+    protected ConnectionDescriptor getConnectionDescriptorFromJwt(RoutingContext ctx, JWT jwt)
+            throws ParseException {
+        String targetId = ctx.pathParam("targetId");
+        // TODO inject the CredentialsManager here to check for stored credentials
```

Comment:
Running `smoketest.sh` gives me a runtime error. What do I need to change to run this PR?
Full logs: http://pastebin.test.redhat.com/1008343
```
Nov 15, 2021 5:54:22 PM io.cryostat.core.log.Logger info
INFO: cryostat started.
Nov 15, 2021 5:54:22 PM io.cryostat.core.log.Logger info
INFO: Selected NoSSL strategy
Nov 15, 2021 5:54:22 PM io.cryostat.core.log.Logger warn
WARNING: No available SSL certificates. Fallback to plain HTTP.
Nov 15, 2021 5:54:22 PM io.cryostat.core.log.Logger info
INFO: Local config path set as /opt/cryostat.d/conf.d
Exception in thread "main" com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	at com.google.gson.Gson.fromJson(Gson.java:870)
	at io.cryostat.configuration.CredentialsManager.lambda$load$2(CredentialsManager.java:85)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:442)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at io.cryostat.configuration.CredentialsManager.load(CredentialsManager.java:86)
	at io.cryostat.Cryostat.main(Cryostat.java:78)
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
	... 17 more
+ cleanup
+ podman pod kill cryostat
```

Reply:
Do you have any local changes applied to `smoketest.sh` or are you just running it as it is in this PR? From the exception trace it looks like your setup is still trying to mount `conf/` to `/opt/cryostat.d/conf.d`, and the `CredentialsManager` is finding something in `/opt/cryostat.d/conf.d/credentials` there that it expects to be JSON but is not.

Do you have a `conf/credentials/cryostat-users.properties` perhaps? That should be `conf/cryostat-users.properties` if so

Reply:
> Do you have a conf/credentials/cryostat-users.properties perhaps? That should be conf/cryostat-users.properties if so

Thanks, that fixed it!